### PR TITLE
Basic schema level validation on models.

### PR DESCRIPTION
**Pull request by parc-jason**
Add in proper schema level validation (so we get 422 not 500 on invalid models).
Add in a business rule check that validates if an identity provider submitted on create form, that it is valid and active.

**Important** I also removed the unique constraint on form.name field.  Either drop your constraint, or reset your database.

# Description

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/src/forms/form/service.js [hunk 1]
```diff
@@ -31,6 +31,14 @@ const service = {
 
       await Form.query(trx).insert(obj);
       if (data.identityProviders && Array.isArray(data.identityProviders) && data.identityProviders.length) {
+        const fips = [];
+        for (const p of data.identityProviders) {
+          const exists = await IdentityProvider.query(trx).where('code', p.code).where('active', true).first();
+          if (!exists) {
+            throw new Problem(422, `${p.code} is not a valid Identity Provider code`);
+          }
+          fips.push({id: uuidv4(), formId: obj.id, code: p.code, createdBy: currentUser.username});
```
Comment (jujaga) [inline app/src/forms/form/service.js:40]: The fips array - is it used anywhere else outside of the loop or the if block? Not seeing it immediately anywhere else in immediate code - could drop the const declaration and push declarations if they're unused.

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wxpbtr-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-form-service-pr-20

File: app/src/forms/common/constants.js [hunk 2]
```diff
@@ -20,5 +20,9 @@ module.exports = Object.freeze({
     FORM_DESIGNER: 'form_designer',
     SUBMISSION_REVIEWER: 'submission_reviewer',
     FORM_SUBMITTER: 'form_submitter'
+  },
+  Regex: {
+    CONFIRMATION_ID: '^[0-9A-Fa-f]{8}$',
+    UUID: '^[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-4[0-9A-Fa-f]{3}-[89ABab][0-9A-Fa-f]{3}-[0-9A-Fa-f]{12}$'
```
Comment (jujaga) [inline app/src/forms/common/constants.js:26]: May be good to name this as UUIDV4 or something as this regex is specifically for version 4.

File: app/src/forms/form/service.js (hunk 1 above)
Comment (parc-jason) [reply app/src/forms/form/service.js:40]: ha, nice one. yeah, it's supposed to be used in the insert...fixed.

**Review by loneil — APPROVED** [review]
Yeah model validations look good

**Pull request merged**